### PR TITLE
EntityImage: Handle unknown elevation exceptions

### LINKED ISSUE #5214: 0.49.19 Nightly from 2/27/2024 - Uncaught Exception when loading saved game causing soft lock
I tried loading a saved game which I loaded before earlier in the game without a problem.  I saved it last night before the movement phase began and tried loading it tonight and got repeated Uncaught Exceptions.  I click OK on an Uncaught Exception box and another one popped up.  This wouldn't stop and the board never loaded, even though the unit display was on a friendly bot-controlled off-board Ballista.  I had to close the Java process via Task Manager.  Log and savegame attached.

[megamek.log](https://github.com/MegaMek/megamek/files/14471320/megamek.log)
[CompanyGame1.sav.gz](https://github.com/MegaMek/megamek/files/14471321/CompanyGame1.sav.gz)

image

### PR DESCRIPTION
**Pull request by SJuliez**
Entity.getElevation throws an exception when the unit has no elevation (e.g. offboard).
Fixes #5214

### CODE REVIEW

**Review by Sleet01 — APPROVED** [review]
lgtm!

**Review by neoancient — APPROVED** [review]

File: megamek/src/megamek/client/ui/swing/tileset/EntityImage.java [hunk 1]
```diff
@@ -197,7 +197,13 @@ public EntityImage(Image base, Image wreck, Camouflage camouflage, Component com
         decal = getDamageDecal(entity, secondaryPos);
         smoke = getSmokeImage(entity, secondaryPos);
         unitHeight = entity.height();
-        unitElevation = entity.getElevation();
+        int elevation = 0;
+        try {
+            elevation = entity.getElevation();
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/tileset/EntityImage.java:202]: Do we really want to use exceptions for flow control here?

**Review by NickAragua — CHANGES_REQUESTED** [review]

File: megamek/src/megamek/client/ui/swing/tileset/EntityImage.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tileset/EntityImage.java]: Doesn't look like the place for an exception. This was old code in Entity. I removed the exception. One could add a hasElevation or knowsElevation method or something but that's a bit beyond this PR.

**Comment by HammerGS** [comment]
Merging as review requests completed.

**Pull request merged**